### PR TITLE
[scala/en] Add traits in objects chapter

### LINKED ISSUE #2759: [scala/en] Request to add "traits"
The [english scala file](https://github.com/adambard/learnxinyminutes-docs/blob/master/scala.html.markdown) is missing the topic "traits" in chapter 5.
```scala
// Traits coming soon!
```
The [german scala file](https://github.com/adambard/learnxinyminutes-docs/blob/master/de-de/scala-de.html.markdown) explains "traits". So the solution could be to translate it :)

```scala
// Traits
// Ähnlich wie Java interfaces, definiert man mit traits einen Objekttyp  
// und Methodensignaturen. Scala erlaubt allerdings das teilweise  
// implementieren dieser Methoden. Konstruktorparameter sind nicht erlaubt.  
// Traits können von anderen Traits oder Klassen erben, aber nur von  
// parameterlosen.  

trait Hund {
	def sorte: String
	def farbe: String
	def bellen: Boolean = true
	def beissen: Boolean
}
class Bernhardiner extends Hund{
	val sorte = "Bernhardiner"
	val farbe = "braun"
	def beissen = false
}

  

scala> b  
res0: Bernhardiner = Bernhardiner@3e57cd70  
scala> b.sorte  
res1: String = Bernhardiner  
scala> b.bellen  
res2: Boolean = true  
scala> b.beissen  
res3: Boolean = false  


// Traits können auch via Mixins (Schlüsselwort "with") eingebunden werden  

trait Bellen {
	def bellen: String = "Woof"
}
trait Hund {
	def sorte: String
	def farbe: String
}
class Bernhardiner extends Hund with Bellen{
	val sorte = "Bernhardiner"
	val farbe = "braun"
}
scala> val b = new Bernhardiner
b: Bernhardiner = Bernhardiner@7b69c6ba
scala> b.bellen
res0: String = Woof
```

### PR DESCRIPTION
**Pull request by andys8**
Fixes #2759 with a translation from german to english.

- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]`
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)

### CODE REVIEW

File: scala.html.markdown [hunk 1]
```diff
@@ -460,8 +460,51 @@ val otherGeorge = george.copy(phoneNumber = "9876")
...
+// Traits can also be used as Mixins (Keyword "with")
+
+trait Bark {
+	def bark: String = "Woof"
+}
+trait Dog {
+	def breed: String
+	def color: String
+}
+class SaintBernard extends Dog with Bark {
```
Comment (vendethiel) [inline scala.html.markdown]: It's probably a good idea to explain that the first one always has to be mixed in with `extends`.

Comment (andys8) [inline scala.html.markdown]: I didn't change semantics when translating from the German version. They should probably in sync.

Comment (vendethiel) [inline scala.html.markdown]: Mh, yeah, probably. Would you mind adding it here and creating an issue for the perl issue later

Comment (andys8) [inline scala.html.markdown]: Sure. I added a short, explaining sentence. Are you fine with it or do you propose an alternative?
I'll backport the changes when this PR get's merged.

**Comment by vendethiel** [comment]
Thanks!

**Comment by andys8** [comment]
Awesome. I created the PR for the original german version with this minor change: #2781

**Pull request merged**